modules/account-lib/src/coin/baseCoin/ed25519KeyPair.ts

```diff
@@ -1,22 +1,95 @@
+import * as nacl from 'tweetnacl';
+import * as hex from '@stablelib/hex';
+import * as stellar from 'stellar-sdk';
+import { Seed, generateAccountFromSeed, generateAccount } from 'algosdk';
 import { BaseKeyPair } from './baseKeyPair';
 import { AddressFormat } from './enum';
-import { NotImplementedError } from './errors';
+import { isPrivateKey, isPublicKey, isSeed, DefaultKeys, KeyPairOptions, ByteKeys } from './iface';
+
+const DEFAULT_SEED_SIZE_BYTES = 32;
 
 export abstract class Ed25519KeyPair implements BaseKeyPair {
-  getAddress(format?: AddressFormat): string {
-    throw new NotImplementedError('getAddress not implemented');
+  protected keyPair: ByteKeys;
+  protected source?: KeyPairOptions;
+
+  /**
+   * Public constructor. By default, creates a key pair with a random master seed.
+   *
+   * @param {KeyPairOptions} source Either a master seed, a private key (extended or raw), or a public key
+   */
+  protected constructor(source?: KeyPairOptions) {
+    let naclKeyPair;
+    if (!source) {
+      const seed = nacl.randomBytes(DEFAULT_SEED_SIZE_BYTES);
+      naclKeyPair = nacl.sign.keyPair.fromSeed(seed);
+      this.setKeyPair(naclKeyPair);
+    } else if (isSeed(source)) {
+      naclKeyPair = nacl.sign.keyPair.fromSeed(new Uint8Array(source.seed));
+      this.setKeyPair(naclKeyPair);
+    } else if (isPrivateKey(source)) {
+      this.recordKeysFromPrivateKey(source.prv);
+    } else if (isPublicKey(source)) {
+      this.recordKeysFromPublicKey(source.pub);
+    } else {
+      throw new Error('Invalid key pair options');
+    }
+  }
+
+  private setKeyPair(naclKeyPair: nacl.SignKeyPair): void {
+    this.keyPair = { prv: naclKeyPair.secretKey, pub: naclKeyPair.publicKey };
   }
 
-  getKeys(): any {
-    // TODO: implement from stellar seed, etc.
-    throw new NotImplementedError('getKeys not implemented');
+  abstract getAddress(format?: AddressFormat): string;
+
+  getKeys(): ByteKeys {
```

Comment:
What do you think @argjv should we return the keys using this format as it is used by the Hedera Tx Builder, or should we encode it to hex and use the DefaultKeys interface?